image/manifest_test.go

```diff
@@ -122,3 +103,45 @@ func TestUnpackLayer(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func createTarFile(name string, list []tarContent) (descriptor, error) {
```

Comment:
> And we have a public Descriptor type in github.com/opencontainers/image-spec/specs-go. Can we return that instead of the private, local descriptor (one step closer to being able to drop the local descriptor ;)?

if it doesn't need to call any of the method attached to `descriptor` - then yes I would use the public `Descriptor` as well - otherwise let's keep `descriptor` for now